packages/odyssey-react-mui/src/labs/ButtonPopover.tsx

```diff
@@ -0,0 +1,81 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {
+  memo,
+  MouseEventHandler,
+  ReactNode,
+  useCallback,
+  useRef,
+  useState,
+} from "react";
+import { Button, ButtonProps } from "../Button";
+import { NullElement } from "../NullElement";
+import { OpenHandle, Popover, PopoverProps } from "./Popover";
+
+export const popoverAlignmentValues = ["left", "right"] as const;
+
+export type ButtonPopoverProps = {
+  /**
+   * The horizontal alignment of the popover.
+   */
+  popoverAlignment?: PopoverProps["popoverAlignment"];
+  /**
+   * The contents of the popover
+   */
+  children: ReactNode | NullElement;
+} & ButtonProps;
+
+const ButtonPopover = (props: ButtonPopoverProps) => {
```

Comment:
Can we name this something like... `ButtonWithPopover` or something similar? Just something that more accurately indicates what this component is/does? The current name leads me to believe that this component is *just* the popover. But, maybe that's just me. Any thoughts?

Reply:
I have zero opinions and am happy to name it whatever. This PR Is mainly to start a conversation (and get design blessing)

Reply:
Got ya, sounds good. Let's go with adaptable `ButtonWithPopover` for now then, if you don't mind :)